packages/amplify_authenticator/example/integration_test/sign_in_force_new_password_test.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:amplify_api/amplify_api.dart';
+import 'package:amplify_authenticator/amplify_authenticator.dart';
+import 'package:amplify_authenticator/src/keys.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+import 'config.dart';
+import 'pages/confirm_sign_in_page.dart';
+import 'pages/sign_in_page.dart';
+import 'pages/test_utils.dart';
+import 'utils/data_utils.dart';
+import 'utils/mock_data.dart';
+
+void main() {
+  final binding = IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+  // resolves issue on iOS. See: https://github.com/flutter/flutter/issues/89651
+  binding.deferFirstFrame();
+
+  final authenticator = MaterialApp(
+    home: Authenticator(
+      child: const SizedBox.shrink(),
+    ),
+  );
+
+  group('Sign In with Force New Password flow', () {
+    final username = generatePhone();
+    final password = generatePassword();
+
+    // Background
+    setUpAll(() async {
+      // Given I'm running the example
+      // "ui/components/authenticator/sign-in-with-phone"
+      await loadConfiguration(
+        'ui/components/authenticator/sign-in-with-phone',
+        additionalConfigs: [AmplifyAPI()],
+      );
+      await adminCreateUser(username, password);
+    });
+
+    tearDownAll(() async {
+      await deleteUser(username);
+    });
+
+    // Scenario: Sign in using a valid phone number and password and user is in
+    // a FORCE_CHANGE_PASSWORD state
+    testWidgets(
+      'Sign in using a valid phone number and password and user is in a '
+      'FORCE_CHANGE_PASSWORD state',
+      (WidgetTester tester) async {
+        final po = SignInPage(tester: tester);
+        await loadAuthenticator(tester: tester, authenticator: authenticator);
+
+        // When I select my country code with status "FORCE_CHANGE_PASSWORD"
+        await po.selectCountryCode();
+
+        // And I type my "phone number" with status "FORCE_CHANGE_PASSWORD"
+        await po.enterUsername(username.substring(2));
```

Comment:
*nit:* It took me a second to understand what `substring(2)` was doing. 
```dart
String phoneWithoutCountryCode = username.substring(2)
await po.enterUsername(phoneWithoutCountryCode);
``` 

might be a little better.

This is pretty picky, so maybe not worth the change.

I think we discussed at some point separating the country code from the rest of the phone number for the purpose of creating a more valuable validator, which would imply that we create some sort of PhoneNumber class which stores them separately. At that point, generatePhoneNumber could probably just return a new PhoneNumber and then it would already be stored separately. So, maybe it could just be changed then.

Reply:
Not nit-picky at all. It should be more clear. Updated - thanks!

And yeah, that would be a good change to have 👍